services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/UploadWithKnownContentLengthHelper.java

```diff
@@ -68,17 +71,20 @@ public UploadWithKnownContentLengthHelper(S3AsyncClient s3AsyncClient,
     }
 
     public CompletableFuture<PutObjectResponse> uploadObject(PutObjectRequest putObjectRequest,
-                                                             AsyncRequestBody asyncRequestBody,
+                                                             Either<AsyncRequestBody, Path> requestBodyOrPath,
```

Comment:
Can we `@Override` `split` in `AsyncFileRequestBody` so that we don't need to do the splitting here?

Reply:
Good idea!